src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final int INDEX_OF_BUSINESS_TYPE_PARAMETER = 0;
+    private static final int INDEX_OF_AMOUNT = 1;
+    private static final int INDEX_OF_SUPPLY_PARAMETER = 0;
+    private static final int INDEX_OF_BUY_PARAMETER = 1;
+    private static final String COMA_SEPARATOR = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] resultParameters = readDataFromFile(fromFileName);
+        int supplyAmount = Integer.parseInt(resultParameters[INDEX_OF_SUPPLY_PARAMETER]);
+        int buyAmount = Integer.parseInt(resultParameters[INDEX_OF_BUY_PARAMETER]);
+        String report = createReport(supplyAmount, buyAmount);
+        writeDataToFile(toFileName, report);
+    }
+
+    private String[] readDataFromFile(String fromFileName) {
+        int supplyAmount = 0;
+        int buyAmount = 0;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            StringBuilder stringBuilder = new StringBuilder();
+            String lineFromFile;
+            while ((lineFromFile = bufferedReader.readLine()) != null) {
+                stringBuilder.append(lineFromFile).append(System.lineSeparator());
+            }
+
+            String[] receivedData = stringBuilder.toString().split(System.lineSeparator());
+            for (String data : receivedData) {
+                String[] dataArray = data.split(COMA_SEPARATOR);
+                if (dataArray[INDEX_OF_BUSINESS_TYPE_PARAMETER].equalsIgnoreCase(SUPPLY)) {
+                    supplyAmount += Integer.parseInt(dataArray[INDEX_OF_AMOUNT]);
+                } else if (dataArray[INDEX_OF_BUSINESS_TYPE_PARAMETER].equalsIgnoreCase(BUY)) {
+                    buyAmount += Integer.parseInt(dataArray[INDEX_OF_AMOUNT]);
+                } else {
+                    System.out.println("Invalid data, skip it");
```

Comment:
Printing to the console is not a good way to handle invalid data within a file reading operation. It would be better to either throw an exception or skip the invalid data silently, depending on the requirements.